src/meta/Dimensions.ts

```diff
@@ -0,0 +1,63 @@
+import { Base } from './Base';
+
+export interface DimensionResults {
```

Comment:
I just cringe a bit when i see `marginTop` (which I think is missing btw?) or `scrollTop`, I can see why they're that way on DOM elements as they specifically do not nest properties. Would it not be immediately obvious what those properties belong to anyway if they were in groups?